lib/ab/configuration.rb

```diff
@@ -0,0 +1,22 @@
+module Ab
+  class << self
+    attr_writer :config
+
+    def config
+      @config ||= Config.new
+    end
+
+    def configure
+      @config ||= Config.new
+      yield(@config)
+    end
+  end
+
+  class Config
+    attr_accessor :logger
+
+    def initialize
+      @logger = Logger.new(nil)
```

Comment:
There are multiple aspects at play here. 

`<oop-zealot-mode-on>`

For one thing being able to pass `nil` to constructor instead of [`LogDevice`](https://github.com/ruby/ruby/blob/ca6b174078fa15f33655be704d9409fdbc4f9929/lib/logger.rb#L568) is implicit knowledge. If you [take](https://github.com/ruby/ruby/blob/ca6b174078fa15f33655be704d9409fdbc4f9929/lib/logger.rb#L317-L320) [a careful](https://github.com/ruby/ruby/blob/ca6b174078fa15f33655be704d9409fdbc4f9929/lib/logger.rb#L366-L368) [look](https://github.com/ruby/ruby/blob/ca6b174078fa15f33655be704d9409fdbc4f9929/lib/logger.rb#L389-L391) into how that works you'll see that `Logger` itself could use a `NullLogDevice` kind of thing since [its implementation/methods are not sure of themselves](https://github.com/avdi/naught#whats-that-supposed-to-mean). But that's not our problem, back to our problem.

So [if we are really serious about performance](https://github.com/vinted/event-tracker/pull/1#discussion_r29657154) then noop is clearly better than [some](https://github.com/ruby/ruby/blob/ca6b174078fa15f33655be704d9409fdbc4f9929/lib/logger.rb#L433-L435) [op](https://github.com/ruby/ruby/blob/ca6b174078fa15f33655be704d9409fdbc4f9929/lib/logger.rb#L364-L368). Also, we [were taught](http://blog.8thlight.com/eric-smith/2011/10/27/thats-not-yours.html) [not to mock](https://github.com/vinted/ab/pull/23/files#diff-cbce597f471835b83022daa27e639b37R23) what we don't own, but in `Loggers` case it's sort of subjective I guess :)

`</oop-zealot-mode-off>`

Just my 2 cents